source/Halibut/Transport/DiscoveryClient.cs

```diff
@@ -27,7 +34,7 @@ public ServiceEndPoint Discover(ServiceEndPoint serviceEndpoint, CancellationTok
             try
             {
                 var log = logs.ForEndpoint(serviceEndpoint.BaseUri);
-                using (var client = TcpConnectionFactory.CreateConnectedTcpClient(serviceEndpoint, log, cancellationToken))
+                using (var client = TcpConnectionFactory.CreateConnectedTcpClient(serviceEndpoint, streamFactory, log, cancellationToken))
```

Comment:
This is a static method.

Reply:
Thought for the future:

🤔 Looking at `TcpConnectionFactory`, it seems to serve 2 purposes. Creating `IConnections` or these static `TcpClients`.

Maybe later we could split this class up appropriately. I.e. move the statics into a `TcpClientFactory` class.

If we do that, then maybe we could even make that something we dependency inject instead.

Again, for later. I just wanted to share my idea 😁

Reply:
Yeah I think re-thinking how this code is laid out would oneday makes sense.

I would encourage us to avoid DI that isn't compile time checked, or DI that makes anything slower or a DI setup that makes it hard to understand the lifetime of things. I guess don't make the code base more complex or less safe by throwing DI at it.